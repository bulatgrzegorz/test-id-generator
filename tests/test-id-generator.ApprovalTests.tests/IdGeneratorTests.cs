using System.Text.Json;
using ApprovalTests;
using ApprovalTests.Reporters;

namespace test_id_generator.ApprovalTests.tests;

[UseReporter(typeof(DiffReporter))]
public class IdGeneratorTests
{
    [Fact]
    public void IdGenerator_PersonVerifyTest()
    {
        Approvals.Verify(new Person(IdGenerator.GetId(), IdGenerator.GetStringId()));
    }
    
    [Fact]
    public void IdGenerator_PersonVerifyTest_2()
    {
        Approvals.Verify(new Person(IdGenerator.GetId(), IdGenerator.GetStringId()));
    }
    
    [Fact]
    public void IdGenerator_ManyIds()
    {
        Approvals.VerifyJson(JsonSerializer.Serialize(new { ints = IdGenerator.GetIds(3), strings = IdGenerator.GetStringIds(6) }));
    }
}

public record Person(int Id, string CorrelationId);