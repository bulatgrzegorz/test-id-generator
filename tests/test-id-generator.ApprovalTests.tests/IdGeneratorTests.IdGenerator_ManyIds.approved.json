{
    "ints": [
        911060166,
        606869171,
        831868293
    ],
    "strings": [
        "9sXw4BG2RtBMqKfI",
        "SRSNBCZ8lkYYHwGC",
        "1KmFdHib42yqP0TQ",
        "tlNZSN7GCZkSNStu",
        "X9WPyKFlJQUKeJsl",
        "AhEwTQUmOPcnwzyl"
    ]
}